docs/basics/ipc.md

```diff
@@ -184,30 +274,62 @@ To detect when a connection is closed from the remote end, override the port clo
 :fontawesome-solid-graduation-cap: 
 [Using `.z`](../kb/using-dotz.md) for more resources, including contributed code for tracing and monitoring
 
-
-## Block, queue, flush
+### Async blocking
 
 To block until any async message is received on handle `h`
 
 ```q
 r:h[] / store message in r
 ```
 
-Messages can be queued for sending to a remote process through using async messaging. kdb+ will queue the serialized message in user space, later writing it to the socket as the remote end drains the message queue. One can see how many messages are queued on a handle and their sizes as a dictionary through the command variable [`.z.W`](../ref/dotz.md#zw-handles "handles").
 
-Sometimes it is useful to send a large number of aysnc messages, but then to block until they have all been sent. This can be achieved through using async flush – invoked as `neg[h][]` or `neg[h](::)`. If you need confirmation that the remote end has received and processed the async messages, chase them with a sync request, e.g. `h""` – the remote end will process the messages on a socket in the order that they are sent.
+## Authentication / Authorization
+
+Basic access control and authentication is supported through using the [-u](cmdline.md#-u-usr-pwd-local)/[-U](cmdline.md#-u-usr-pwd) command-line option to specify a file of users and passwords.
+
+In order to provide further customizations of the authentication process, [.z.pw](../ref/dotz.md#zpw-validate-user) callback is called immediately after successful –u/-U authentication (if specified at startup – otherwise .z.pw is the first authentication check done by a kdb+ process). The ability to set .z.pw to user defined function, allows allows integration with enterprise standards such as LDAP, Kerberos, OpenID Connect,etc
+
+Finer grained authorization can be implemented by tracking user info with active handles, and customizing sync/async callbacks for user-level permissioning e.g. server with protected funcs for sync calls
+
+```q
+q)\p 5000
+q)allowedFns:(`func1;`func2;`func3;+;-) / list of allowed function/ops to call
+q)checkFn:{if[not x in allowedFns;'(-3!x)," not allowed"];}
+q)validatePT:{if[0h=t:type x;if[(not 0h=type first x)&1=count first x;checkFn first x;];.z.s each x where 0h=type each x;];}
+q).z.pg:{if[10h=type x;x:parse x;];validatePT x;eval x}
+```
+
+client trying to access protected functions:
+```q
+q)h:hopen 5000
+q)h"1+1"
+2
+q)h"1*1"
+'* not allowed
+  [0]  h"1*1"
+       ^
+```
+
+!!! note
+    ticker plants, and other high-volume message sources such as feed handlers, will most likely be inserting data via .z.ps. To cater for such high volumes, the handles of those processes should be used to avoid the overhead of these validation checks. That is, feeds and tickerplants could be viewed as trusted processes
```

Comment:
```suggestion
    Ticker plants and other high-volume message sources, such as feed handlers, generally insert data using `.z.ps`. To manage such high volumes, the handles of those processes should be used to avoid the overhead of these validation checks. That is, feeds and tickerplants could be viewed as trusted processes.
```